javascript_nodejs/50.contoso-cafe-bot/dialogs/main/index.js

```diff
@@ -0,0 +1,260 @@
+                        break;
+                    }
+                }
+                case DialogTurnStatus.waiting: {
+                    // The active dialog is waiting for a response from the user, so do nothing
+                    break;
+                }
+                case DialogTurnStatus.cancelled: {
+                    // The active dialog's stack has been cancelled
+                    await dc.context.sendActivity(MessageFactory.suggestedActions(GenSuggestedQueries(), `Is there anything else I can help you with?`));
+                    // End active dialog
+                    await dc.cancelAll();
+                    break;
+                }
+            }
+            dialogTurnResult = (dialogTurnResult === undefined) ? new TurnResultHelper(DialogTurnStatus.empty) : dialogTurnResult;
+            return dialogTurnResult;
+        }
+        /**
+         * Method to begin appropriate child dialog based on user input
+         * 
+         * @param {Object} dc 
+         * @param {Object} onTurnProperty 
+         * @param {Object} childDialogPayload 
+         */
+        async beginChildDialog(dc, onTurnProperty, childDialogPayload) {
+            switch (onTurnProperty.intent) {
+                // Help, ChitChat and QnA share the same QnA Maker model. So just call the QnA Dialog.
+                case QnADialog.Name: 
+                case ChitChatDialog.Name: 
+                case HelpDialog.Name: {
+                    return await dc.begin(QnADialog.Name);
+                }
+                case CancelDialog.Name: {
+                    return await dc.begin(CancelDialog.Name, childDialogPayload);
+                } case WhoAreYouDialog.Name: {
+                    return await this.beginWhoAreYouDialog(dc, onTurnProperty);
+                } case FindCafeLocationsDialog.Name: {
+                    return await dc.begin(FindCafeLocationsDialog.Name);
+                } case WhatCanYouDoDialog.Name: {
+                    return await this.beginWhatCanYouDoDialog(dc, onTurnProperty);
+                } case NONE_INTENT: {
+                    await dc.context.sendActivity(`I'm still learning.. Sorry, I do not know how to help you with that.`);
+                    return await dc.context.sendActivity(`Follow [this link](https://www.bing.com/search?q=${dc.context.activity.text}) to search the web!`);
+                }
+            }
+        }
+        /**
+         * Method to evaluate if the requested user operation is possible.
+         * User could be in the middle of a multi-turn dialog where intteruption might not be possible or allowed.
+         * 
+         * @param {Object} dc 
+         * @param {String} requestedOperation 
+         * @returns {Object} outcome object
+         */
+        async isRequestedOperationPossible(dc, requestedOperation) {
```

Comment:
function is `async` but does not `await` anything. Javascript allows you to `await` non thenable functions so you can be confident you will never throw awaiting on a non promise:
```js
function synchronousFunction() { return true; }
async function doStuff() {
  await synchronousFunction();
}
```